src/gicv3.rs

```diff
@@ -402,6 +486,83 @@ impl GicV3 {
     pub fn sgi_ptr(&mut self) -> *mut SGI {
         self.sgi
     }
+
+    fn gicd_barrier(&mut self) {
+        // SAFETY: We know that `self.gicd` is a valid and unique pointer to the registers of a
+        // GIC distributor interface.
+        unsafe {
+            while (&raw const (*self.gicd).ctlr)
+                .read_volatile()
+                .contains(GicdCtlr::RWP)
+            {}
+        }
+    }
+
+    fn gicd_modify_control(&mut self, flags: GicdCtlr, op: BitOp) {
```

Comment:
What would you think of this instead taking a function that updates the flags? Something like

`fn gicd_modify_control(&mut self, f: impl FnOnce(GicdCtlr) -> GicdCtlr)`